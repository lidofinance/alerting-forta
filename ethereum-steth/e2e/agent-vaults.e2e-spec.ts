import { configureContainer, Finding } from "forta-agent";
import { AwilixContainer, asFunction } from "awilix";
import {
  provideAgentPath,
  provideRunBlock,
  provideRunTransaction,
} from "./utils";

const TEST_TIMEOUT = 60_000; // ms

describe("agent-vaults e2e tests", () => {
  let runBlock: (
    blockHashOrNumber: string | number,
    initBlock?: number,
    skipInit?: boolean,
  ) => Promise<Finding[]>;
  let runTransaction: (txHash: string) => Promise<Finding[]>;
  let logSpy: jest.SpyInstance;

  beforeAll(() => {
    logSpy = jest.spyOn(console, "log");
    logSpy.mockImplementation(() => {});
  });

  beforeEach(async () => {
    const container = configureContainer() as AwilixContainer;
    container.register({
      agentPath: asFunction(provideAgentPath("subagents/vaults/agent-vaults")),
      runTransaction: asFunction(provideRunTransaction),
      runBlock: asFunction(provideRunBlock),
    });

    // https://docs.forta.network/en/latest/cli/#invoke-commands-programmatically
    runTransaction = container.resolve("runTransaction");
    runBlock = container.resolve("runBlock");
  });

  afterAll(() => {
    jest.resetAllMocks();
  });

  it(
    "should process block with high EL vault balance difference",
    async () => {
      let findings = await runBlock(17007842, 17007842);
      expect(findings).toMatchSnapshot();
    },
    TEST_TIMEOUT,
  );
});
