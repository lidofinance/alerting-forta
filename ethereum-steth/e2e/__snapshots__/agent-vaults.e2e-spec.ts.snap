// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`agent-vaults e2e tests should process block with high EL vault balance difference 1`] = `
Array [
  Finding {
    "addresses": Array [],
    "alertId": "EL-VAULT-BALANCE-CHANGE",
    "description": "EL Vault Balance has increased by 689.017 ETH",
    "labels": Array [],
    "metadata": Object {},
    "name": "ðŸ’µ EL Vault Balance significant change",
    "protocol": "ethereum",
    "severity": 1,
    "type": 4,
  },
]
`;
